runtime/v2/manager.go

```diff
@@ -140,14 +141,73 @@ func (m *MM) ValidateGenesis(genesisData map[string]json.RawMessage) error {
 	return nil
 }
 
-// InitGenesis performs init genesis functionality for modules.
-func (m *MM) InitGenesis() {
-	panic("implement me")
+// ExportGenesis performs export genesis functionality for modules
+func (m *MM) ExportGenesis(ctx context.Context) (map[string]json.RawMessage, error) {
+	return m.ExportGenesisForModules(ctx, []string{})
 }
 
-// ExportGenesis performs export genesis functionality for modules
-func (m *MM) ExportGenesis() {
-	panic("implement me")
+// ExportGenesisForModules performs export genesis functionality for modules
+func (m *MM) ExportGenesisForModules(ctx context.Context, modulesToExport []string) (map[string]json.RawMessage, error) {
+	if len(modulesToExport) == 0 {
+		modulesToExport = m.config.ExportGenesis
+	}
+	// verify modules exists in app, so that we don't panic in the middle of an export
+	if err := m.checkModulesExists(modulesToExport); err != nil {
+		return nil, err
+	}
+
+	type genesisResult struct {
+		bz  json.RawMessage
+		err error
+	}
+
+	channels := make(map[string]chan genesisResult)
+	for _, moduleName := range modulesToExport {
+		mod := m.modules[moduleName]
+		if module, ok := mod.(appmodulev2.HasGenesis); ok {
+			channels[moduleName] = make(chan genesisResult)
+			go func(module appmodulev2.HasGenesis, ch chan genesisResult) {
+				jm, err := module.ExportGenesis(ctx)
+				if err != nil {
+					ch <- genesisResult{nil, err}
+					return
+				}
+				ch <- genesisResult{jm, nil}
+			}(module, channels[moduleName])
+		} else if module, ok := mod.(sdkmodule.HasABCIGenesis); ok {
+			channels[moduleName] = make(chan genesisResult)
+			go func(module sdkmodule.HasABCIGenesis, ch chan genesisResult) {
+				jm, err := module.ExportGenesis(ctx)
+				if err != nil {
+					ch <- genesisResult{nil, err}
+				}
+				ch <- genesisResult{jm, nil}
+			}(module, channels[moduleName])
+		}
+	}
+
+	genesisData := make(map[string]json.RawMessage)
+	for moduleName := range channels {
+		res := <-channels[moduleName]
```

Comment:
nit: this is blocking on the first channel in the map. If you would move the module name into the `genesisResult`, you can have a channel just for the results instead that would be a fifo. Not sure if this gives some significant performance benefit or makes it harder to read. 🤷